core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -343,6 +343,15 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     value
   }
 
+  /** Control our logLevel. This overrides any user-defined log settings.
+   * @param logLevel The desired log level as a string.
+   * Valid log levels include: ALL, DEBUG, ERROR, FATAL, INFO, OFF, TRACE, WARN
+   * Invalid log level defaults to DEBUG
```

Comment:
Yes - I'm suggesting that before you pass anything to log4j we make sure it's one of the valid logging levels. Therefore, the default behavior of log4j's `toLevel` will be irrelevant.